FirebaseRemoteConfig/Tests/SwiftAPI/RemoteConfigConsole.swift

```diff
@@ -0,0 +1,244 @@
+  /// remote config, if it exists
+  public var activeRemoteConfig: [String: Any]? {
+    var config: [String: Any]?
+    perform(configRequest: .get) { latestConfigJSON in
+      config = latestConfigJSON
+    }
+    if let config = config {
+      save(config)
+    }
+    return config
+  }
+
+  /// Exposing this initializer allows us to
+  /// create`RemoteConfigConsole` instances without
+  /// dependence on a`GoogleService-Info.plist`.
+  init(projectID: String) {
+    self.projectID = projectID
+    syncWithConsole()
+  }
+
+  /// This initializer will attempt to read from a
+  /// `GoogleService-Info.plist` to set `projectID`
+  convenience init() {
+    let currentBundle = Bundle(for: type(of: self))
+    let projectID = currentBundle.plistValue(for: "PROJECT_ID", from: "GoogleService-Info.plist")
+    self.init(projectID: projectID! as! String)
+  }
+
+  // MARK: - Public API
+
+  public func updateRemoteConfig(parameters: [String: Any]) {
+    var updatedConfig: [String: Any] = latestConfig
+
+    let latestParameters = latestConfig["parameters"] as? [String: Any]
+    if var updatedParameters = latestParameters {
+      for (key, value) in parameters {
+        updatedParameters.updateValue(["defaultValue": ["value": value]], forKey: key)
+      }
+      updatedConfig.updateValue(updatedParameters, forKey: "parameters")
+
+    } else {
+      var newParameters = [String: Any]()
+
+      for (key, value) in parameters {
+        newParameters.updateValue(["defaultValue": ["value": value]], forKey: key)
+      }
+      updatedConfig.updateValue(newParameters, forKey: "parameters")
+    }
+
+    publish(config: updatedConfig)
+  }
+
+  public func updateRemoteConfigValue(_ value: Any, for key: String) {
+    var updatedConfig: [String: Any] = latestConfig
+
+    let latestParameters = latestConfig["parameters"] as? [String: Any]
+    if var parameters = latestParameters {
+      parameters.updateValue(["defaultValue": ["value": value]], forKey: key)
+      updatedConfig.updateValue(parameters, forKey: "parameters")
+
+    } else {
+      updatedConfig.updateValue([key: ["defaultValue": ["value": value]]], forKey: "parameters")
+    }
+
+    publish(config: updatedConfig)
+  }
+
+  public func removeRemoteConfigValue(for key: String) {
+    var updatedConfig: [String: Any] = latestConfig
+
+    let latestParameters = latestConfig["parameters"] as? [String: Any]
+    if var parameters = latestParameters {
+      parameters.removeValue(forKey: key)
+      updatedConfig.updateValue(parameters, forKey: "parameters")
+    }
+
+    publish(config: updatedConfig)
+  }
+
+  public func clearRemoteConfig() {
+    var updatedConfig: [String: Any]! = latestConfig
+    updatedConfig.removeValue(forKey: "parameters")
+    publish(config: updatedConfig)
+  }
+
+  // Other possible API's
+  public func currentRemoteConfigValue(for key: String) -> Any { "" }
+
+  public func validateRemoteConfig(with values: [String: Any]) -> Bool { false }
```

Comment:
I was thinking about some other API's that'd be useful for testing. Like the rest, these two would be synchronous. Since remote config can contain JSON values instead of simple primitive key value pairs, I was thinking of a  `validateRemoteConfig` that takes in a dictionary representing a JSON remote config value. Then validates that against the current remote config JSON value stored on the console.

Reply:
SGTM!

Reply:
I played around with getting these to work but was running into trouble trying to pick out the config value from the remote config HTTPS response object and infer its type (string, number, bool). I think it's probably best to just use the regular Firebase API `fetchConfig` for retrieving the currently active config value and then comparing it to values/dictionaries accordingly since the Firebase unwraps the config value nicely already.